packages/mobile/src/qrcode/QRScanner.tsx

```diff
@@ -23,19 +23,35 @@ interface DispatchProps {
   handleBarcodeDetected: typeof handleBarcodeDetected
 }
 
+interface State {
+  isScanningEnabled: boolean
+}
+
 type Props = DispatchProps & WithNamespaces & NavigationFocusInjectedProps
 
-class QRScanner extends React.Component<Props> {
+class QRScanner extends React.Component<Props, State> {
   static navigationOptions = () => ({
     ...headerWithBackButton,
     headerTitle: i18n.t('sendFlow7:scanCode'),
   })
 
+  state = {
+    isScanningEnabled: true,
+  }
+
   camera: RNCamera | null = null
 
+  // This method would be called multiple times with the same
+  // QR code, so we need to catch only the first one
   onBardCodeDetected = (rawData: any) => {
-    Logger.debug('QRScanner', 'Bar code detected')
-    this.props.handleBarcodeDetected(rawData)
+    if (this.state.isScanningEnabled) {
+      this.setState({ isScanningEnabled: false })
+      Logger.debug('QRScanner', 'Bar code detected')
+      this.props.handleBarcodeDetected(rawData)
```

Comment:
You should move this code to the callback function of setState, otherwise I still think this can run more than once. 
Also please clear the timeout in the componentWillUnmount just in case

Reply:
Have addressed that